azurelinuxagent/common/telemetryevent.py

```diff
@@ -82,33 +82,66 @@ def __eq__(self, other):
         return isinstance(other, TelemetryEventParam) and other.name == self.name and other.value == self.value
 
 
+# too-few-public-methods<R0903> Disabled: This class is used as an Enum
+class TelemetryEventPriorities(object): # pylint: disable=R0903
+    """
+    Class defining the priorities for telemetry events. Lower the number, higher the priority
+
+    Note: 0 is reserved for a feature like QuickLog in the Windows Agent (i.e. the ability to send out telemetry
+    instantly rather than waiting for a minute for the monitor thread to pick up the events)
+    """
+    AGENT_EVENT = 1 # Agent events always get the highest priority
+    EXTENSION_EVENT_NEW_PIPELINE = 2    # Prioritize extensions using the new dedicated pipeline over extensions hijacking the agent pipeline
+    EXTENSION_EVENT_OLD_PIPELINE = 3
+
+
 class TelemetryEvent(DataContract):
-    def __init__(self, eventId=None, providerId=None):
+    def __init__(self, eventId=None, providerId=None, priority=TelemetryEventPriorities.AGENT_EVENT):
         self.eventId = eventId # pylint: disable=C0103
         self.providerId = providerId # pylint: disable=C0103
         self.parameters = DataContractList(TelemetryEventParam)
         self.file_type = ""
+        self._priority = priority
 
     # Checking if the particular param name is in the TelemetryEvent.
     def __contains__(self, param_name):
         return param_name in [param.name for param in self.parameters]
 
+    def __le__(self, other):
```

Comment:
Introduced total_ordering for all events based on their priority. This helps with fetching the Events in the PriorityQueue in the correct order

Reply:
nice!

Reply:
Removed the priority queue logic as of now, will introduce later if extension telemetry starts blocking agent events drastically